File: platforms/ios/Dezel/Dezel/View/TextInputObserver.swift
/**
 * @protocol TextInputObserver
 * @since 0.7.0
 */
public protocol TextInputObserver: AnyObject {

	/**
	 * @method didChange
	 * @since 0.7.0
	 */
	func didChange(textInput: TextInput, value: String)

	/**
	 * @method didFocus
	 * @since 0.7.0
	 */
	func didFocus(textInput: TextInput)

	/**
	 * @method didBlur
	 * @since 0.7.0
	 */
	func didBlur(textInput: TextInput)

}

File: platforms/android/Dezel/dezel/src/main/java/ca/logaritm/dezel/view/TextInputObserver.kt
package ca.logaritm.dezel.view

/**
 * @interface TextInputObserver
 * @since 0.7.0
 */
public interface TextInputObserver {

	/**
	 * @method onChange
	 * @since 0.7.0
	 */
	fun onChange(textInput: TextInput, value: String)

	/**
	 * @method onFocus
	 * @since 0.7.0
	 */
	fun onFocus(textInput: TextInput)

	/**
	 * @method onBlur
	 * @since 0.7.0
	 */
	fun onBlur(textInput: TextInput)

}
